chaco/plots/scatterplot_1d.py

```diff
@@ -32,7 +32,9 @@ class ScatterPlot1D(Base1DPlot):
     marker = MarkerTrait
 
     # The pixel size of the marker, not including the thickness of the outline.
-    marker_size = Float(4.0)
+    marker_size = Union(Float, Int,
+                        Array(Float), Array(Int),
+                        default_value=None)
```

Comment:
Why changing the defaut to None? The 4.0 value was a good default. By changing the default, you now have to update the `_render` method.

The upstream issue was related to the fact that only floats were supported. Using a Union type, you now allow other data type, which is solving the problem. Adding a tests that reproduces the uptsream problem would be good

Reply:
I somehow remembered that using 4.0 as default will raise an error, but after retest, works fine... Will use 4.0 instead.

Reply:
@homosapien-lcy adding a new manual example is not adding a test! 
As discussed, tests are based on the `unittest.TestCase` in order too be picked up by the CI and shoud be added here: https://github.com/enthought/chaco/blob/main/chaco/plots/tests/test_scatterplot_1d.py